hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -1594,7 +1598,9 @@ public OzoneInputStream getKey(
             .setDataSize(keyInfo.getDataSize())
             .setCreationTime(keyInfo.getCreationTime())
             .setModificationTime(keyInfo.getModificationTime())
-            .setReplicationConfig(keyInfo.getReplicationConfig())
+            .setReplicationConfig(replicationConfig instanceof ECReplicationConfig
+                    ? RatisReplicationConfig.getInstance(HddsProtos.ReplicationFactor.ONE)
+                    : keyInfo.getReplicationConfig())
```

Comment:
Could you elaborate why did you choose to set the replication factor to ONE and RATIS in case of EC? You can do this in the method javadoc and in the PR description. You can add why and how the length is calculated in case of EC. Thanks!